lib/internal/event_target.js

```diff
@@ -973,6 +834,98 @@ ObjectDefineProperties(NodeEventTarget.prototype, {
 
 // EventTarget API
 
+/**
+ * @see https://dom.spec.whatwg.org/#dictdef-eventlisteneroptions
+ * @typedef {Object} AddEventListenerOptions
+ * @property {null} __proto__
+ * @property {boolean} [capture=false]
+ * @property {boolean} passive
+ * @property {boolean} [once=false]
+ * @property {import('./abort_controller').AbortSignal|null} signal
+ */
+
+/**
+ * @see https://dom.spec.whatwg.org/#event-flatten-more
+ * @param {*} options
+ * @returns {AddEventListenerOptions}
+ */
+function flattenMoreOptions (options) {
+  if (typeof options === 'boolean') {
+    return {
+      __proto__: null,
+      capture: options,
+      passive: false,
+      once: false,
+      signal: null 
+    }
+  }
+
+  validateDictionary(options, 'options')
+
+  const validatedOptions = {
+    __proto__: null,
+    passive: Boolean(options.passive),
+    once: Boolean(options.once),
```

Comment:
you already have a options variable, and you're re-creating it. This is unnecessary copying.

Reply:
This is essentially implementing a webidl dictionary converter, we do this in undici. Options could also be a proxy & these could be readonly getters so we can't safely overwrite properties on the object